DetectorDescription/DDCMS/plugins/dd4hep/DDDefinitions2Objects.cc

```diff
@@ -875,11 +950,26 @@ void Converter<DDLPosPart>::operator()(xml_h element) const {
 
   if (!parent.isValid() && strchr(parentName.c_str(), NAMESPACE_SEP) == nullptr)
     parentName = ns.prepend(parentName);
-  parent = ns.volume(parentName);
+  parent = ns.volume(parentName, false);
+  if (parentPtr == nullptr)
+    parentPtr = ns.getVolPtr(parentName);
+  if (!parent.isValid() && parentPtr == nullptr)
+    throw runtime_error("Unknown volume identifier:" + parentName);
 
   if (!child.isValid() && strchr(childName.c_str(), NAMESPACE_SEP) == nullptr)
     childName = ns.prepend(childName);
   child = ns.volume(childName, false);
+  if (childPtr == nullptr)
+    childPtr = ns.getVolPtr(childName);
+  if (childPtr != nullptr && parentPtr != nullptr && ((*parentPtr)->IsAssembly() || (*childPtr)->IsAssembly())) {
+    printout(ns.context()->debug_placements ? ALWAYS : DEBUG,
+             "DD4CMS",
+             "***** Placing assembly parent %s, child %s",
+             parentName.c_str(),
+             childName.c_str());
+    placeAssembly(parentPtr, parentName, childPtr, childName, copy, transform, ns);
```

Comment:
@cvuosalo - please, let dd4hep place it:
https://cmssdt.cern.ch/lxr/source/Geometry/EcalCommonData/plugins/dd4hep/DDEcalBarrelNewAlgo.cc#0976